### ISSUE TITLE
Announcing selections from skill picker control

### ISSUE THREAD

**Issue by gobyrne** [GCTC-NTGC/gc-digital-talent#7320]
## 🕵️  Details
> When using the existing skill selector to add essential or asset skills, the UI should announce automatically that it has been added immediately without me having to move focus away and back again onto the button to check.
 
## 🌟 Acceptance Criteria

- [ ] Add a toast notification that is presented when the user adds a skill (using skill picker). Toast should say "{skillName} added".
- [ ] Do this for new skill picker only (`SkillDialog`)

### Original issue info

**AT Configuration**

NVDA with Chrome on Windows 

**Tester Feedback** 

 - "When using the existing skill selector to add essential or asset skills, the UI should announce automatically that it has been added immediately without me having to move focus away and back again onto the button to check."

**💻 Related WCAG Criteria**

 - 1.3.1: Info and Relationships: Information, [structure](https://w3c.github.io/wcag/understanding/info-and-relationships#dfn-structure), and [relationships](https://w3c.github.io/wcag/understanding/info-and-relationships#dfn-relationships) conveyed through [presentation](https://w3c.github.io/wcag/understanding/info-and-relationships#dfn-presentation) can be [programmatically determined](https://w3c.github.io/wcag/understanding/info-and-relationships#dfn-programmatically-determined) or are available in text.

- 1.3.3. Sensory Characteristics: Instructions provided for understanding and operating content do not rely solely on sensory characteristics of components such as shape, color, size, visual location, orientation, or sound.

 - 2.1.1 Keyboard: All [functionality](https://www.w3.org/WAI/WCAG21/Understanding/keyboard#dfn-functionality) of the content is operable through a [keyboard interface](https://www.w3.org/WAI/WCAG21/Understanding/keyboard#dfn-keyboard-interface) without requiring specific timings for individual keystrokes, except where the underlying function requires input that depends on the path of the user's movement and not just the endpoints.

## 🧪 Testing Steps

1. Log in to the Poll operator UI using NVDA, navigate to admin and select **"Pools."**
2. Activate **"Create a new pool."**
3. Start a blank job poster for an IT-01 Helpdesk Support in the DCM.
4. On the Edit pool page, skip to the **Essential skills** section.
5. Use the selectors to add a skill.
6. Observe if NVDA notifies you if you added a skill.
 

**Related Issues**

- https://github.com/GCTC-NTGC/gc-digital-talent/issues/4874
- https://github.com/GCTC-NTGC/gc-digital-talent/issues/4794
- https://github.com/GCTC-NTGC/gc-digital-talent/issues/5487
